### PR TITLE
feat: use curses

### PR DESCRIPTION
**Pull request by MyGodIsHe**
Hi Tom,

I have prepared a patch for using the [curses](https://docs.python.org/3/howto/curses.html) library. I know that you do not want to support two versions, but I tried to simplify it as much as possible. Please take a look.

Best Regards
Ilya

### CODE REVIEW

File: gping/pinger.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
 #!/usr/bin/env python
 # coding=utf8
 
+import curses
```
Comment (orf) [inline gping/pinger.py]: Can you wrap this in a try/except so it doesn't fail on Windows?

File: gping/pinger.py [hunk 2]
```diff
@@ -102,9 +102,150 @@ def box(self, bottom_left_corner, top_right_corner, paint=None, blank=False):
...
+    def create_canvas(self, width, height):
+        # We space for a newline after each line, so restrict the hight/width a bit
+        width, height = width - 1, height - 1  # Our work area is slightly smaller
+        canvas = Canvas(width, height)
+        # Draw a box around the edges of the screen, one character in.
+        canvas.box(
+            Point(1, 1), Point(width - 1, height - 1)
+        )
+        # We use islice to slice the data because we can't do a ranged slice on a dequeue :(
+        data_slice = list(islice(self.buff, 0, width - 3))
```
Comment (orf) [inline gping/pinger.py]: Is `list()` required here? We only iterate it with `enumerate()`

File: gping/pinger.py [hunk 3]
```diff
@@ -121,103 +262,54 @@ def lines(self):
...
+        current_color = None
+        for top, line in enumerate(canvas.data):
+            for left, pair in enumerate(line):
+                char, color = pair
+                if color != current_color and char != " ":
+                    current_color = color
+                curses_color = self.COLOR_MAP[current_color]
+                try:
+                    self.stdscr.addstr(top, left, char, curses_color)
+                except:
```
Comment (orf) [inline gping/pinger.py]: `except Exception:`?

Comment (MyGodIsHe) [reply gping/pinger.py]: Thanks. Got it.

File: gping/pinger.py (hunk 2 above)
Comment (MyGodIsHe) [reply gping/pinger.py]: Yes, that is right. I will fix it.

File: gping/pinger.py (hunk 1 above)
Comment (MyGodIsHe) [reply gping/pinger.py]: Ok

**Comment by orf** [comment]
Thanks! I will give this a test locally when I can, then merge 👍

**Comment by MyGodIsHe** [comment]
By the way. I tried to run the code from the master branch in windows 10 and it does not work. Maybe I did something wrong. It falls on bad encoding. I tried to fix it, but still does not work. Black screen.

**Comment by MyGodIsHe** [comment]
ping C:

**Comment by orf** [comment]
I’ve switched the project to Rust, and incorporated some of these changes. Thanks for the pr!

**Pull request closed without merging**